src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3001 @@
+				}
+			}
+			goto passToOthers; // For exiting the UI mode, I think
+		}
+	}
+
+	//if holding horizontal encoder button down and pressing back clear automation
+	//if you're on automation overview, clear all automation
+	//if you're in the automation editor, clear the automation for the parameter in focus
+
+	else if (b == BACK && currentUIMode == UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON) {
+		//only allow clearing of a clip if you're in the automation overview
+		if (on && clip->lastSelectedParamID == kNoLastSelectedParamID) {
+			goto passToOthers;
+		}
+		else if (on && clip->lastSelectedParamID != kNoLastSelectedParamID) {
+			//delete automation of current parameter selected
+
+			char modelStackMemory[MODEL_STACK_MAX_SIZE];
+			ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+
+			ModelStackWithAutoParam* modelStackWithParam =
+			    getModelStackWithParam(modelStack, clip, clip->lastSelectedParamID, clip->lastSelectedParamKind);
+
+			if (modelStackWithParam && modelStackWithParam->autoParam) {
+				Action* action = actionLogger.getNewAction(ACTION_AUTOMATION_DELETE, false);
+				modelStackWithParam->autoParam->deleteAutomation(action, modelStackWithParam);
+
+				numericDriver.displayPopup(HAVE_OLED ? "Automation deleted" : "DELETED");
+				setDisplayParameterNameTimer();
+			}
+		}
+	}
+
+	//Select encoder
+	//if you're not pressing shift and press down on the select encoder, toggle interpolation on/off
+
+	else if (!Buttons::isShiftButtonPressed() && b == SELECT_ENC) {
+		if (on) {
+			if (interpolation == RuntimeFeatureStateToggle::Off) {
+
+				interpolation = RuntimeFeatureStateToggle::On;
+
+				numericDriver.displayPopup(HAVE_OLED ? "Interpolation On" : "ON");
+			}
+			else {
+				interpolation = RuntimeFeatureStateToggle::Off;
+
+				numericDriver.displayPopup(HAVE_OLED ? "Interpolation Off" : "OFF");
+			}
+			setDisplayParameterNameTimer();
+		}
+	}
+
+	//when you press affect entire in a kit, the parameter selection needs to reset
+	else if (b == AFFECT_ENTIRE) {
+		initParameterSelection();
+		goto passToOthers;
+	}
+
+	else {
+passToOthers:
+
+		uiNeedsRendering(this);
+
+		ActionResult result = InstrumentClipMinder::buttonAction(b, on, inCardRoutine);
+		if (result != ActionResult::NOT_DEALT_WITH) {
+			return result;
+		}
+
+		result = ClipView::buttonAction(b, on, inCardRoutine);
+
+		setDisplayParameterNameTimer();
+
+		return result;
+	}
+
+	if (on && (b == KEYBOARD || b == CLIP_VIEW || b == SESSION_VIEW)) {
+		numericDriver.cancelPopup();
+	}
+
+	if (on && (b != KEYBOARD && b != CLIP_VIEW && b != SESSION_VIEW)) {
+		setDisplayParameterNameTimer();
+		uiNeedsRendering(this);
+	}
+
+	return ActionResult::DEALT_WITH;
+}
+
+//simplified version of the InstrumentClipView::enterScaleMode function. No need to render any animation.
+void AutomationInstrumentClipView::enterScaleMode(uint8_t yDisplay) {
```

Comment:
@bobtwinkles great catch on the enter/exit scale mode. I removed the excess code that was really only there to animate changes to notes which I don't display in my view.

That said, these functions are still required for the purpose of adjusting the scroll position of the sidebar piano roll when you enter/exit scale mode.